### PR TITLE
feat: move confirmation modals to prompt

### PR DESCRIPTION
**Pull request by el**
### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/request/comment.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { useMutation } from 'react-query';
+import { DELETE_COMMENT_MUTATION } from '../graphql/comments';
+import { EmptyResponse } from '../graphql/emptyResponse';
+import { Post } from '../graphql/posts';
+import { usePostComment } from '../hooks/usePostComment';
+import { useRequestProtocol } from '../hooks/useRequestProtocol';
+import { apiUrl } from '../lib/config';
+
+export const deleteComment = ({
```
Comment (el) [inline packages/shared/src/request/comment.ts:9]: I didnt want to make the main components bloated, so I created this folder/files to keep the request helpers

File: packages/shared/src/request/post.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import request from 'graphql-request';
+import { EmptyResponse } from '../graphql/emptyResponse';
+import { BAN_POST_MUTATION, DELETE_POST_MUTATION } from '../graphql/posts';
+import { apiUrl } from '../lib/config';
+
+export const banPost = (id: string): Promise<EmptyResponse> => {
```
Comment (el) [inline packages/shared/src/request/post.ts:6]: I didnt want to make the main components bloated, so I created this folder/files to keep the request helpers

File: packages/shared/src/components/post/PostComments.tsx [hunk 3]
```diff
@@ -116,6 +112,23 @@ export function PostComments({
...
+      description:
+        'Are you sure you want to delete your comment? This action cannot be undone.',
+      okButton: {
+        title: 'Delete',
+        className: 'btn-primary-ketchup',
+      },
+    };
+    if (await showPrompt(options)) {
+      await deleteComment({ commentId, parentId, post });
+    }
```
Comment (rebelchris) [inline packages/shared/src/components/post/PostComments.tsx:130]: Would it not be neater to attach this to the actually button?
The cancel doesn't go to this resolve?

File: packages/shared/src/request/comment.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  parentId,
+  post,
+}: {
+  commentId: string;
+  parentId: string | null;
+  post: Post;
+}): Promise<EmptyResponse> => {
+  const { deleteCommentCache } = usePostComment(post);
+  const { requestMethod } = useRequestProtocol();
+  const { mutateAsync: deleteCommentRequest } = useMutation<EmptyResponse>(
```
Comment (rebelchris) [inline packages/shared/src/request/comment.ts:20]: Perhaps we can move this to the `graphQL` `comments` file like you did previously to have everything uniform?

File: packages/shared/src/request/post.ts (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/request/post.ts:6]: Technically also could sit in the graphQL layer right?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Did you migrate all the existing test to the new flow?

File: packages/shared/src/request/post.ts (hunk 2 above)
Comment (el) [reply packages/shared/src/request/post.ts:6]: Yeap. I am fine with both. Either here or in graphql. Just dont want to bloat the component

Comment (rebelchris) [reply packages/shared/src/request/post.ts:6]: Quite liked the fact graphQL had all the functions related to it.
So maybe we should just standardize it 🙏

Comment (el) [reply packages/shared/src/request/post.ts:6]: Makes sense, done 👍

File: packages/shared/src/components/post/PostComments.tsx (hunk 3 above)
Comment (el) [reply packages/shared/src/components/post/PostComments.tsx]: this is not a callback. `showPrompt` returns `true` for okButton, and `false` for the cancel.

**Review by rebelchris — APPROVED** [review]
Thanks for the cleanup 🥳

**Pull request merged**